### PR TITLE
[#1721] A bit of cleanup and autoplay working

### PR DESCRIPTION
**Pull request by dseif**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/86a408ff6ff2237/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/86a408ff6ff2237/output.txt

Total script time: 0.97 mins

Server is running on http://pr847.butyr.org/templates/basic/
Run the Test Suite: http://pr847.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/49d9e76360308df/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/49d9e76360308df/output.txt

Total script time: 0.81 mins

Server is running on http://pr847.butyr.org/templates/basic/
Run the Test Suite: http://pr847.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/8d33073c88e1af3/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/8d33073c88e1af3/output.txt

Total script time: 0.58 mins

Server is running on http://pr847.butyr.org/templates/basic/
Run the Test Suite: http://pr847.butyr.org/test/

File: src/embed.js [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
 function init( window, document ) {
 
+  var stateClasses = [
+    "embed-playing",
+    "embed-paused",
+    "embed-ended"
+  ];
+
   /**
-   * embed.js is a separate, top-level entry point into the requirejs
```
Comment (jbuck) [inline src/embed.js]: don't change this alignment

File: src/embed.js [hunk 2]
```diff
@@ -50,21 +64,23 @@ function init( window, document ) {
...
-      shareSize = $( "share-size" ).value.split( "x" ),
-      width = shareSize[0],
-      height = shareSize[1];
+        // Sizes are strings: "200x400"
+        shareSize = $( "share-size" ).value.split( "x" ),
+        width = shareSize[ 0 ],
+        height = shareSize[ 1 ];
+
     return '<iframe src="' + src + '" width="' + width + '" height="' + height +
-           '" frameborder="0" mozallowfullscreen webkitallowfullscreen allowfullscreen></iframe>';
```
Comment (jbuck) [inline src/embed.js]: don't change this alignment

File: src/embed.js [hunk 3]
```diff
@@ -148,44 +158,54 @@ function init( window, document ) {
     ],
     function( URI, Controls, TextboxWrapper ) {
       /**
-       * Expose Butter so we can get version info out of the iframe doc's embed.
```
Comment (jbuck) [inline src/embed.js]: Don't change this alignment

File: src/embed.js [hunk 4]
```diff
@@ -148,44 +158,54 @@ function init( window, document ) {
...
+          qs = URI.parse( window.location.href ).queryKey,
+          container = document.querySelectorAll( ".container" )[ 0 ],
+          videoContainer = document.getElementById( "video-container" ),
+          controlsElement = document.getElementById( "controls" ),
+          _container,
+          autoHideTimeout,
+          hide = true;
 
       /**
-       * The embed can be configured via the query string:
```
Comment (jbuck) [inline src/embed.js:161]: Don't change this alignment

File: src/embed.js [hunk 5]
```diff
@@ -148,44 +158,54 @@ function init( window, document ) {
...
+      *   fullscreen = 1{default}|0    whether to allow fullscreen mode (e.g., hide/show button)
+      *   loop       = 1|0{default}    whether to loop when hitting the end
+      *   branding   = 1{default}|0    whether or not to show the Mozilla Popcorn branding
+      *   showinfo   = 1{default}|0    whether to show video title, author, etc. before playing
+      **/
       config = {
         autohide: qs.autohide === "0" ? false : true,
         autoplay: qs.autoplay === "1" ? true : false,
         controls: qs.controls === "0" ? false : true,
-        start: qs.start|0,
```
Comment (jbuck) [inline src/embed.js]: Leave this change; what it does is Math.floor() whatever start is.

Comment (dseif) [inline src/embed.js]: That's awesome, never knew that. Thanks!

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/3b9e6b76e84485a/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/cb8ddccdcf523ca/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/3b9e6b76e84485a/output.txt

Total script time: 1.47 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/cb8ddccdcf523ca/output.txt

Total script time: 0.55 mins

Server is running on http://pr847.butyr.org/templates/basic/
Run the Test Suite: http://pr847.butyr.org/test/

File: css/controls.css [hunk 6]
```diff
@@ -5,6 +5,15 @@
   background: white;
   border: 1px solid #d5d6d5;
   height: 41px;
+  visibility: visible;
+  opacity: 1;
+  transition: opacity .5s ease-in-out;
+  -moz-transition: opacity .5s ease-in-out;
+  -webkit-transition: opacity .5s ease-in-out;
```
Comment (jbuck) [inline css/controls.css:12]: Lint is complaining about the fact that you're not including -o- and -ms- prefixes

File: src/embed.js (hunk 5 above)
Comment (jbuck) [inline src/embed.js]: Revert this change you made here; you don't want the default for end to be 0

File: src/embed.js [hunk 7]
```diff
@@ -196,28 +214,144 @@ function init( window, document ) {
...
+      if ( config.controls ) {
+        popcorn.controls( true );
         Controls( "controls", popcorn );
         show( "#controls" );
       }
 
-      setupClickHandlers( popcorn, config );
-      setupEventHandlers( popcorn, config );
+      // Setup UI based on config options
+      if ( !config.branding) {
```
Comment (cadecairos) [inline src/embed.js]: missing a space here.

File: css/controls.css [hunk 8]
```diff
@@ -5,6 +5,15 @@
   background: white;
   border: 1px solid #d5d6d5;
   height: 41px;
+  visibility: visible;
+  opacity: 1;
+  transition: opacity .5s ease-in-out;
+  -moz-transition: opacity .5s ease-in-out;
```
Comment (cadecairos) [inline css/controls.css:11]: Still getting lint complaints about the -o- prefix:

### Linting CSS files

css/controls.css: line 11, col 3, Error - The property -o-transition is compatible with -moz-transition and -webkit-transition and should be included as well.
css/controls.css: line 11, col 3, Error - The property -ms-transition is compatible with -moz-transition and -webkit-transition and should be included as well.

File: src/embed.js [hunk 9]
```diff
@@ -227,10 +361,10 @@ function init( window, document ) {
 }
 
 // Source tree case vs. require-built case.
-if( typeof require === "undefined" ) {
+if ( typeof require === "undefined" ) {
   Popcorn.getScript( "../../external/require/require.js", function() {
     init( window, window.document );
- });
+});
```
Comment (cadecairos) [inline src/embed.js]: indentation needed.
